### PR TITLE
ARIA 1.2 Combobox support

### PR DESCRIPTION
**Pull request by tombrunet**
This pull request updates rule checking support from ARIA 1.1 guidance to ARIA 1.2 guidance for comboboxes.

### CODE REVIEW

**Comment by tombrunet** [comment]
@cpandhi or @sharitrewin Can you please review the help / messages for these new combobox rules?

* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_activedescendant.mdx
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_autocomplete_inline.mdx
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_focusable_elements.mdx
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_haspopup.mdx
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_popup_reference.mdx
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/help/combobox_version.mdx

You'll find the messages for the six combobox rules on line 943:
* https://github.com/IBMa/equal-access/blob/combobox-1.2/accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts#L934

Help is fairly similar to the previous combobox help.

**Comment by cpandhi** [comment]
@tombrunet suggested change to 2 error message

File: accessibility-checker-engine/help/combobox_activedescendant.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+or grid, that can dynamically pop up to help the user set the value of the input. When a combobox 
+receives focus, the focus should be set on the text `<input>` element. Unless using a `dialog`
+popup, the `aria-activedescendant` must be used to identify the currently focused selection in the
+popup, so people who use assistive technologies are able accurately navigate and interact with the content.
+
+<div id="locSnippet"></div>
+
+### What to do
+
+* Unless the `has-popup` atttribute value is `dialog`, set the active descendant focus using the `aria-activedescendant` only on the combobox’s `<input>` element.
```
Comment (sharitrewin) [inline accessibility-checker-engine/help/combobox_activedescendant.mdx:34]:
```suggestion
* Unless the `has-popup` atttribute value is `dialog`, set the active descendant focus using the `aria-activedescendant` attribute only on the combobox’s `<input>` element.
```

File: accessibility-checker-engine/help/combobox_activedescendant.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+<Row>
+<Column colLg={11} colMd={5} colSm={4} className="toolMain">
+
+### Why is this important?
+
+A combobox is a UI interactive component that combines an input and another element, such as a listbox
+or grid, that can dynamically pop up to help the user set the value of the input. When a combobox 
+receives focus, the focus should be set on the text `<input>` element. Unless using a `dialog`
+popup, the `aria-activedescendant` must be used to identify the currently focused selection in the
+popup, so people who use assistive technologies are able accurately navigate and interact with the content.
```
Comment (sharitrewin) [inline accessibility-checker-engine/help/combobox_activedescendant.mdx:28]:
```suggestion
popup, so people who use assistive technologies are able to accurately navigate and interact with the content.
```

File: accessibility-checker-engine/help/combobox_focusable_elements.mdx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+</Row>
+<Row>
+<Column colLg={11} colMd={5} colSm={4} className="toolMain">
+
+### Why is this important?
+
+A combobox is a UI interactive component that combines an input and another element, such as a listbox
+or grid, that can dynamically pop up to help the user set the value of the input. When a combobox 
+receives focus, the focus should be set on the text `<input>` element that can either display the 
+current value or supports editing, so people who use assistive technologies are able accurately
```
Comment (sharitrewin) [inline accessibility-checker-engine/help/combobox_focusable_elements.mdx:27]:
```suggestion
current value or supports editing, so people who use assistive technologies are able to accurately 
```

File: accessibility-checker-engine/help/combobox_haspopup.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+</Column>
+</Row>
+<Row>
+<Column colLg={11} colMd={5} colSm={4} className="toolMain">
+
+### Why is this important?
+
+An element with `role="combobox"` may have a popup element with the role of "listbox", "tree", "grid", or "dialog".
+If the value for the `aria-haspopup` attribute does not match the value for the popup element's `role` 
+attribute, people who use assistive technologies may may see unexpected behavior and will not know how
```
Comment (sharitrewin) [inline accessibility-checker-engine/help/combobox_haspopup.mdx:26]:
```suggestion
attribute, people who use assistive technologies may see unexpected behavior and will not know how 
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 5]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
+        "Fail_1.0_missing_owns": "The 'aria-owns' attribute of the expanded combobox is missing",
+        "Fail_1.2_missing_controls": "The 'aria-controls' attribute of the expanded combobox is missing",
+        "Fail_1.0_popup_reference_missing": "The 'aria-owns' attribute \"{0}\" of the expanded combobox does not reference a valid popup 'id' value",
+        "Fail_1.2_popup_reference_missing": "The 'aria-controls' attribute \"{0}\" of the expanded combobox does not reference a valid popup 'id' value",
+        "Fail_combobox_expanded_hidden": "The combobox 'aria-expanded' attribute is true, but the combobox popup is not visible",
+        "Fail_combobox_collapsed_visible": "The combobox 'aria-expanded' attribute is false, but the combobox popup is visible"
+    },
+    "combobox_haspopup": {
+        0: "The combobox attribute 'aria-haspopup' value must be appropriate for the role of the element referenced by `aria-controls` (WAI-ARIA 1.2) or `aria-owns` (WAI-ARIA 1.0)",
```
Comment (sharitrewin) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:952]: If I remember correctly, we cannot use the backquote in these messages, so the quotes around aria-controls and aria-owns might need to be changed.

**Review by sharitrewin — CHANGES_REQUESTED** [review]
These new rules look great. I suggested a few minor tweaks to the help, and also flagged a possible issue in the index.ts file with the use of backquote.

**Comment by cpandhi** [comment]
@tombrunet suggested feedback on the error message for clarity:
"combobox_autocomplete": {
"Pass": "The combobox 'aria-autocomplete' attribute is correctly set on the text input element with a valid value"
"Fail_1": "The combobox 'aria-autocomplete' attribute is incorrectly set on an element within the popup referenced by \"{0}\", should be set on the text input element"

  "combobox_popup_reference": {
"Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
 This seems a little confusing to me as a pass message. Should we say
"Pass_collapsed": "The combobox in its collapsed state does not have any reference to a visible popup as required",

"combobox_focusable_elements": {
        0: "Tabbable focus for the combobox must be allowed only on the text input, except when using a dialog popup",
        "Pass": "DOM focus is allowed only on the combobox element as required",
        "Fail_not_tabbable": "DOM focus is not allowed on the combobox element",
        "Fail_tabbable_child": "The popup of the combobox has DOM focus or has 'aria-activedescendant' defined which is not allowed"
    },

"combobox_active_descendant": {
"Fail_active_role_invalid": "The 'aria-activedescendant' \"{0}\" references an element with the roles \"{1}\", which does not have a valid WAI-ARIA role of 'option', 'gridcell', 'row', or 'treeitem'",

"combobox_autocomplete": {
        0: "A combobox that supports autocompletion behavior must have the 'aria-autocomplete' attribute only on its text input element with a valid value, a value '\"inline\"' is not supported",
        "Pass": "The combobox approprietly does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
        "Fail_1": "The combobox has the 'aria-autocomplete' attribute incorrectly set on an element within the popup referenced by \"{0}\"",
        "Fail_inline": "The combobox does not support an 'aria-autocomplete' attribute set to 'inline'"
    },

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 6]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+    "combobox_version": {
+        0: "The combobox design pattern must be valid for WAI-ARIA 1.2",
+        "Pass_1.0": "The combobox design pattern is detected as WAI-ARIA 1.0, which is allowed by WAI-ARIA 1.2",
+        "Fail_1.1": "The combobox design pattern is detected as WAI-ARIA 1.1, which is not allowed by WAI-ARIA 1.2",
+        "Pass_1.2": "The combobox design pattern is detected as WAI-ARIA 1.2"
+    },
+    "combobox_popup_reference": {
+        0: "The 'aria-controls' (for WAI-ARIA 1.2) or the 'aria-owns' (for WAI-ARIA 1.0) attribute of the expanded combobox must reference a valid popup 'id' value",
+        "Pass_expanded": "The combobox popup referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) exists and is visible",
+        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:943]:
```suggestion
        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
```suggestion
        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
```suggestion
        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
```suggestion
        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
```suggestion
        "Pass_collapsed": "The combobox popup is not referenced by 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) or is hidden",
```
```suggestion
        "Pass_collapsed": "The combobox in its collapsed state does not reference any visible popup as required",
```

Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:943]:
```suggestion
        "Pass_collapsed": "The combobox popup in its collapsed state does not reference any visible popup as required",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts (hunk 5 above)
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:952]:
```suggestion
        0: "The combobox attribute 'aria-haspopup' value must be appropriate for the role of the element referenced by `aria-controls` (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0)",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 7]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_combobox_collapsed_visible": "The combobox 'aria-expanded' attribute is false, but the combobox popup is visible"
+    },
+    "combobox_haspopup": {
+        0: "The combobox attribute 'aria-haspopup' value must be appropriate for the role of the element referenced by `aria-controls` (WAI-ARIA 1.2) or `aria-owns` (WAI-ARIA 1.0)",
+        "Pass": "The 'aria-controls' (WAI-ARIA 1.2) or 'aria-owns' (WAI-ARIA 1.0) appropriately references a valid popup 'id' value",
+        "Fail_popup_role_invalid": "The 'role' value \"{0}\" of the popup element \"{1}\" should be one of \"listbox\", \"grid\", \"tree\" or \"dialog\"",
+        "Fail_combobox_popup_role_mismatch": "The value of the combobox 'aria-haspopup' attribute \"{0}\" does not match the 'role' value of the popup element \"{1}\""
+    },    
+    "combobox_focusable_elements": {
+        0: "Tabbable focus for the combobox must be allowable only on the text input, except when using a dialog popup",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:958]:
```suggestion
        0: "Tabbable focus for the combobox must be allowable only on the text input, except when using a dialog popup",
```
```suggestion
        0: "Tabbable focus for the combobox must be allowed only on the text input, except when using a dialog popup",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 8]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Pass": "DOM focus is allowed on the combobox element and is not allowed on the popup",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:959]:
```suggestion
        "Pass": "DOM focus is allowed only on the combobox element as required",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 9]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_not_tabbable": "DOM focus is not allowed on the combobox",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:960]:
```suggestion
        "Fail_not_tabbable": "The combobox element does not have DOM focus as required",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 10]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_tabbable_child": "The popup of the combobox allows DOM focus or has 'aria-activedescendant' defined"
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:961]:
```suggestion
        "Fail_tabbable_child": "The popup of the combobox has DOM focus or has 'aria-activedescendant' defined which is not allowed"
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 11]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+    },
+    "combobox_active_descendant": {
+        0: "'aria-activedescendant' must be used to define focus within the combobox popup, except when using a dialog popup",
+        "Pass": "'aria-activedescendant' is used appropriately for this combobox",
+        "Fail_missing": "The element referenced by 'aria-activedescendant' \"{0}\" does not exist",
+        "Fail_not_in_popup": "The element referenced by 'aria-activedescendant' \"{0}\" does not exist within the popup referenced by 'id' \"{1}\"",
+        "Fail_active_role_invalid": "The 'aria-activedescendant' \"{0}\" references an element with the roles \"{1}\", which does not have a WAI-ARIA role of 'option', 'gridcell', 'row', or 'treeitem'",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:968]:
```suggestion
        "Fail_active_role_invalid": "The 'aria-activedescendant' \"{0}\" references an element with the roles \"{1}\", which does not have a valid WAI-ARIA role of 'option', 'gridcell', 'row', or 'treeitem'",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 12]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_active_not_selected": "The 'aria-activedescendant' \"{0}\" references an element that does not have 'aria-selected' set to true",
+    },
+    "combobox_autocomplete": {
+        0: "A combobox that supports autocompletion behavior must not have an 'aria-autocomplete' attribute value of '\"inline\"' and must have the 'aria-autocomplete' attribute only on its text input element",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:972]:
```suggestion
        0: "A combobox that supports autocompletion behavior must have the 'aria-autocomplete' attribute only on its text input element with a valid value, a value of '\"inline\"' is not supported",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 13]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Pass": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:973]:
```suggestion
        "Pass": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 14]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_1": "The combobox has the 'aria-autocomplete' attribute set on an element within the popup referenced by \"{0}\"",
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:974]:
```suggestion
        "Fail_1": "The combobox has the 'aria-autocomplete' attribute incorrectly set on an element within the popup referenced by \"{0}\"",
```

File: accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts [hunk 15]
```diff
@@ -931,47 +931,48 @@ let a11yNls = {
...
+        "Fail_inline": "The combobox does not support an 'aria-autocomplete' attribute set to 'inline'"
```
Comment (cpandhi) [inline accessibility-checker-engine/src/v2/checker/accessibility/nls/index.ts:975]:
```suggestion
        "Fail_inline": "The combobox does not support an 'aria-autocomplete' attribute value set to '\"inline\"' "
```

**Comment by tombrunet** [comment]
@cpandhi @sharitrewin Accepted all of the inline suggestions. Anything else?

**Comment by cpandhi** [comment]
@tombrunet Everything else looks good to me.

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example1_collapsed.html [hunk 16]
```diff
@@ -44,7 +44,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                     },
                     "reasonId": "Pass",
-                    "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                    "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example1_collapsed.html:47]:
```suggestion
                    "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example1_expanded.html [hunk 17]
```diff
@@ -43,7 +43,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                     },
                     "reasonId": "Pass",
-                    "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                    "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example1_expanded.html:46]:
```suggestion
                    "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example2_collapsed.html [hunk 18]
```diff
@@ -208,7 +208,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example2_collapsed.html:211]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example2_expanded.html [hunk 19]
```diff
@@ -208,7 +208,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example2_expanded.html:211]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example3_collapsed.html [hunk 20]
```diff
@@ -208,7 +208,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example3_collapsed.html:211]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example3_expanded.html [hunk 21]
```diff
@@ -208,7 +208,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example3_expanded.html:211]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example4_collapsed.html [hunk 22]
```diff
@@ -73,7 +73,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example4_collapsed.html:76]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example4_expanded.html [hunk 23]
```diff
@@ -73,7 +73,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example4_expanded.html:76]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example5_collapsed.html [hunk 24]
```diff
@@ -36,7 +36,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example5_collapsed.html:39]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

File: accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example5_expanded.html [hunk 25]
```diff
@@ -36,7 +36,7 @@
                     "aria": "/document[1]/main[1]/combobox[1]"
                 },
                 "reasonId": "Pass",
-                "message": "The combobox does not use 'aria-autocomplete' value 'inline' and does not have 'aria-autocomplete' defined within the popup",
+                "message": "The combobox does not use 'aria-autocomplete' value '"inline"' nor does it have 'aria-autocomplete' defined within the popup",
```
Comment (sharitrewin) [inline accessibility-checker-engine/test/v2/checker/accessibility/rules/combobox_autocomplete_ruleunit/w3c_example5_expanded.html:39]:
```suggestion
                "message": "The combobox does not use 'aria-autocomplete' value '\"inline\"' nor does it have 'aria-autocomplete' defined within the popup",
```

**Review by sharitrewin — CHANGES_REQUESTED** [review]
It looks like there are a few " characters inside strings that need an escape character.

**Comment by tombrunet** [comment]
@sharitrewin I think I merged all of those...

**Review by sharitrewin — APPROVED** [review]
Looks good!

**Pull request merged**